Mathlib/NumberTheory/LucasPrimality.lean

```diff
@@ -39,23 +39,12 @@ group must itself have order `p-1`, which only happens when `p` is prime.
 -/
 theorem lucas_primality (p : ℕ) (a : ZMod p) (ha : a ^ (p - 1) = 1)
     (hd : ∀ q : ℕ, q.Prime → q ∣ p - 1 → a ^ ((p - 1) / q) ≠ 1) : p.Prime := by
-  have h0 : p ≠ 0 := by
-    rintro ⟨⟩
-    exact hd 2 Nat.prime_two (dvd_zero _) (pow_zero _)
-  have h1 : p ≠ 1 := by
-    rintro ⟨⟩
-    exact hd 2 Nat.prime_two (dvd_zero _) (pow_zero _)
-  have hp1 : 1 < p := lt_of_le_of_ne h0.bot_lt h1.symm
-  have order_of_a : orderOf a = p - 1 := by
-    apply orderOf_eq_of_pow_and_pow_div_prime _ ha hd
-    exact tsub_pos_of_lt hp1
-  haveI : NeZero p := ⟨h0⟩
+  have h : p ≠ 0 ∧ p ≠ 1 := by
```

Comment:
If you want, you could do
```suggestion
  obtain ⟨h0, h1⟩ : p ≠ 0 ∧ p ≠ 1 := by
```

Reply:
I use the `p ≠ 0 ∧ p ≠ 1` hypothesis as is for `hp1`, so I think it's fine as is.